### PR TITLE
feat: TV/Fullscreen mode for dashboards

### PR DESCRIPTION
**Pull request by filp**
### Description:
- closes: #3195 
- Adds a TV/Fullscreen mode for dashboards:

image

- Hides navbar when in fullscreen mode:
image

Additionally, correctly handles browser-native fullscreen controls (e.g if the user exits fullscreen manually, we also disable TV mode in Lightdash).

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by joaoviana** [comment]
Relates to #3195

File: packages/frontend/src/components/common/Dashboard/DashboardHeader.tsx [hunk 1]
```diff
@@ -267,7 +273,32 @@ const DashboardHeader = ({
                 <PageActionsContainer>
                     {userCanExportData && <DashboardRefreshButton />}
 
-                    {!!userCanManageDashboard && (
+                    {!isEditMode && document.fullscreenEnabled && (
```
Comment (joaoviana) [inline packages/frontend/src/components/common/Dashboard/DashboardHeader.tsx:276]: can we use 
```suggestion
                    {!isEditMode && isFullScreen && (
``` 
?

File: packages/frontend/src/pages/Dashboard.tsx [hunk 2]
```diff
@@ -273,6 +275,40 @@ const Dashboard: FC = () => {
...
+        };
+
+        document.addEventListener('fullscreenchange', onFullscreenChange);
+
+        return () =>
+            document.removeEventListener(
+                'fullscreenchange',
+                onFullscreenChange,
+            );
+    });
```
Comment (joaoviana) [inline packages/frontend/src/pages/Dashboard.tsx:310]: this is great, but we can also use this v useful hook 😉 https://mantine.dev/hooks/use-fullscreen/

**Review by joaoviana — APPROVED** [review]
Works great! 

Another thing to consider (can be on a separate PR):
1. Go to dashboard
2. Enter full screen mode
3. Click `Explore from here` on a chart tile
4. User gets redirected to a chart view 
5. Full screen remains

Not too sure on what we expect the behaviour to be like but I would probably exit Full screen mode on route change 👍

File: packages/frontend/src/components/common/Dashboard/DashboardHeader.tsx (hunk 1 above)
Comment (filp) [reply packages/frontend/src/components/common/Dashboard/DashboardHeader.tsx:276]: The check on `document.fullscreenEnabled` is used to hide the button if fullscreen isn't available in this user agent, despite the super confusing name 😵‍💫

File: packages/frontend/src/pages/Dashboard.tsx (hunk 2 above)
Comment (filp) [reply packages/frontend/src/pages/Dashboard.tsx:310]: Awesome, thanks for the heads up -- I keep getting surprised by features built into mantine 😄 I'll refactor with this hook.

**Comment by filp** [comment]
Merging as-is and following up with small improvements in another PR - using `useFullscreen` from mantine is actually not super straight-forward since it introduces some wonky behavior (tl;dr: the extra indirection compared to this implementation can cause render-loop-related issues when figuring out if/when to go/exit fullscreen).

**Pull request merged**